src/libs/API/commands/BaseAPICommandBlocking.js

```diff
@@ -0,0 +1,44 @@
+import BaseAPICommand from './BaseAPICommand';
+import Onyx from 'react-native-onyx';
+import ONYXKEYS from '../../../ONYXKEYS';
+import CONST from '../../../CONST';
+
+/**
+ * API Commands that need to have some kind of blocking UI while the network request is being made
+ * can extend this class.
+ */
+export default class extends BaseAPICommand {
+    // Use withOnyx to access the data about this blocking request (eg. ValidateEmailCommand.blockingCommandOnyxKey)
+    static blockingCommandOnyxKey = `BlockingCommand-${this.commandName}`;
+
+    /**
+     * An abstract method to define what should happen before the network request is made
+     */
+    startBlocking() {
+        Onyx.merge(this.blockingCommandOnyxKey, {loading: true, error: null});
+    }
+
+    /**
+     * An abstract method to define what should happen after the network request is made
+     */
+    finishBlocking() {
+        Onyx.merge(this.blockingCommandOnyxKey, {loading: false});
```

Comment:
**possible issue:** it looks like `finishBlocking()` stops the loading and is called when the request promise resolves - yet the data might not be available via Pusher. So, should a request be considered "finished" when the `fetch()` call resolves or when new data appears?

Reply:
Oh, good point. I guess it would make sense to move this into the data-changed handler.